### PR TITLE
23547 uxa suggested changes

### PR DESCRIPTION
**Pull request by avni-work**
*Issue:*

- bcgov/entity/issues/23547

*Description of changes:*

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by avni-work** [comment]
gcbrun

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-253-7pntljjc.web.app

File: strr-web/components/bcros/form-section/property/Form.vue [hunk 1]
```diff
@@ -260,7 +260,8 @@ const validateHostResidence = () => {
 }
 
 const validateNumberOfRoomsForRent = () => {
-  if (!formState.propertyDetails.numberOfRoomsForRent || formState.propertyDetails.numberOfRoomsForRent <= 0) {
+  if (!formState.propertyDetails.numberOfRoomsForRent || formState.propertyDetails.numberOfRoomsForRent < 1) {
+    formState.propertyDetails.numberOfRoomsForRent = 1
```
Comment (dimak1) [inline strr-web/components/bcros/form-section/property/Form.vue:264]: So you can decrease number of rooms to less than 1?

**Comment by dimak1** [comment]
On the Review page, the bottom values should not be in `flex-row`, instead included in the `grid` with the rest of the fields.

Screenshot 2024-10-31 at 11 16 11

**Review by dimak1 — CHANGES_REQUESTED** [review]
Small changes on the Review page.

File: strr-web/components/bcros/form-section/property/Form.vue (hunk 1 above)
Comment (avni-work) [reply strr-web/components/bcros/form-section/property/Form.vue:264]: no the user cannot decrement past 1

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-253-7pntljjc.web.app

**Comment by avni-work** [comment]
> On the Review page, the bottom values should not be in `flex-row`, instead included in the `grid` with the rest of the fields.
> 
> Screenshot 2024-10-31 at 11 16 11

i believe this now done! please verify

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-253-7pntljjc.web.app

**Comment by andyyanggov** [comment]
✅ Order of the fields is correct
✅ User can manually enter a number for 'Number of Rooms for Rent'

File: strr-web/components/bcros/form-section/property/Form.vue (hunk 1 above)
Comment (dimak1) [reply strr-web/components/bcros/form-section/property/Form.vue]: Then we don't need this check and set to 1 ?

**Comment by avni-work** [comment]
image

File: strr-web/components/bcros/form-section/review/Form.vue [hunk 2]
```diff
@@ -89,15 +91,13 @@
                   }}
                 </p>
               </BcrosFormSectionReviewItem>
+            </div>
+            <div class="flex flex-row justify-between w-full desktop:mb-[24px] mobile:flex-col">
```
Comment (dimak1) [inline strr-web/components/bcros/form-section/review/Form.vue:95]: Please move this `BcrosFormSectionReviewItem` into the grid div.

**Comment by dimak1** [comment]
The row is still there @avni-work

**Comment by rstens** [comment]
> ✅ Order of the fields is correct ✅ User can manually enter a number for 'Number of Rooms for Rent'

Do we want to put an upper limit on the number of rooms? The field in the db has a max of 2147483647

**Comment by rstens** [comment]
A bit of an odd error message:
image

**Comment by rstens** [comment]
Mandatory field address not showing "Required":
image

**Comment by rstens** [comment]
Clcik on listing details and then move to other field fills the field with: about:blank
This will fail in the submit.

image

**Review by rstens — CHANGES_REQUESTED** [review]
We could use a test-data-id on the Number of Rooms input.

<input data-v-4510615f="" type="number" class="w-full text-center outline-none border-none" min="1">

**Comment by avni-work** [comment]
> A bit of an odd error message: image

fixed

**Comment by avni-work** [comment]
> Mandatory field address not showing "Required": image

won't fix here as did not change this field as part of this ticket

**Review by dimak1 — COMMENTED** [review]
No further comments - just unit test fixes.

**Comment by avni-work** [comment]
> Click on listing details and then move to other field fills the field with: about:blank This will fail in the submit.
> 
> image
> 
> App will flag it: image
> 
> But it is then impossible to leave the field blank, because it will revert to the about:blank content.

won't fix this here either, as not part of ticket but happy to fix in another ticket

**Comment by rstens** [comment]
> We could use a test-data-id on the Number of Rooms input.
> 
> do i just add the id or want me to write up a test too?

just the test id

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-253-7pntljjc.web.app

**Comment by rstens** [comment]
> > Mandatory field address not showing "Required": image
> 
> won't fix here as did not change this field as part of this ticket

ok

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-253-7pntljjc.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-253-7pntljjc.web.app

**Comment by avni-work** [comment]
image

**Comment by avni-work** [comment]
image

**Comment by dimak1** [comment]
If @andyyanggov approves the Review pages, I'm good to approve the PR. Thanks!

**Comment by andyyanggov** [comment]
Honestly, for the purpose of the MVP, I'm not overly concerned about how the key-value pairs are displayed on the Review page. I think it may be too much effort to get 100% with the time we have (especially for something that might change post-MVP). So I'm good with this. 👍

**Comment by dimak1** [comment]
Thanks for input @andyyanggov

**Review by dimak1 — APPROVED** [review]
LG! Thanks for the tests.

**Comment by avni-work** [comment]
i will let @rstens also look through this, so don't have to open a new pr later

**Comment by rstens** [comment]
One more:
image

**Comment by avni-work** [comment]
> One more: image

good catch, forgot to save the message in en.json. pushed a change

**Comment by avni-work** [comment]
image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-253-7pntljjc.web.app

**Review by rstens — APPROVED** [review]

**Comment by rstens** [comment]
Let's go!!!!

**Comment by rstens** [comment]
image

**Pull request merged**